pkg/lib/scoped/token_retriever.go

```diff
@@ -81,11 +81,11 @@ func getAPISecret(logger logrus.FieldLogger, kubeclient operatorclient.ClientInt
 // specific ServiceAccount via annotations kubernetes.io/service-account.name
 func filterSecretsBySAName(saName string, secrets *corev1.SecretList) map[string]*corev1.Secret {
 	secretMap := make(map[string]*corev1.Secret)
-	for _, ref := range secrets.Items {
+	for i, ref := range secrets.Items {
 		annotations := ref.GetAnnotations()
 		value := annotations[corev1.ServiceAccountNameKey]
 		if value == saName {
-			secretMap[ref.Name] = &ref
+			secretMap[ref.Name] = &secrets.Items[i]
```

Comment:
We _could_ use a deepCopy here:
* Pro: Object in the client cache isn't returned. As it is, users should never edit the secrets returned by this function for risk of concurrent map writes. Also prevents changes to these secrets if updated on cluster.
* Con: Memory footprint.

Reply:
I'd vote for DeepCopy here since we potentially leave ourselves open to race conditions in the future where the secrets list gets changed on cluster in the middle of this operation. In that case, we could output a reference that is invalid. How big of an extra memory footprint are we thinking here?

Reply:
The canonical solution I've seen (and typically use) for this is:
```go
for _, ref := range secrets.Items {
	ref := ref
	...
}

Reply:
I'm for this.

Reply:
Updated per @joelanford's suggestion, but moved it inside the if statement.